rct229/rulesets/ashrae9012019/section5/section5rule4.py

```diff
@@ -0,0 +1,79 @@
+    get_surface_conditioning_category_dict,
+)
+
+
+class Section5Rule4(RuleDefinitionListIndexedBase):
+    """Rule 4 of ASHRAE 90.1-2019 Appendix G Section 5 (Envelope)"""
+
+    def __init__(self):
+        super(Section5Rule4, self).__init__(
+            rmrs_used=UserBaselineProposedVals(False, True, False),
+            required_fields={
+                "$": ["weather"],
+                "weather": ["climate_zone"],
+            },
+            each_rule=Section5Rule4.BuildingRule(),
+            index_rmr="baseline",
+            id="5-4",
+            description="Baseline roof assemblies must conform with assemblies detailed in Appendix A",
+            ruleset_section_title="Envelope",
+            standard_section="Section G3.1-5(b) Building Envelope Modeling Requirements for the Baseline building",
+            is_primary_rule=False,
+            list_path="ruleset_model_instances[0].buildings[*]",
+            data_items={"climate_zone": ("baseline", "weather/climate_zone")},
+        )
+
+    class BuildingRule(RuleDefinitionListIndexedBase):
+        def __init__(self):
+            super(Section5Rule4.BuildingRule, self).__init__(
+                rmrs_used=UserBaselineProposedVals(False, True, False),
+                required_fields={"$": "building_segments"},
+                each_rule=Section5Rule4.BuildingRule.SurfaceRule(),
+                index_rmr="baseline",
+                list_path="$..surfaces[*]",
+            )
+
+        def create_data(self, context, data=None):
+            building = context.baseline
+            return {
+                "surface_conditioning_category_dict": get_surface_conditioning_category_dict(
+                    data["climate_zone"], building
+                ),
+            }
+
+        def list_filter(self, context_item, data):
+            surface = context_item.baseline
+            return get_opaque_surface_type(surface) == OST.ROOF
+
+        class SurfaceRule(PartialRuleDefinition):
+            def __init__(self):
+                super(Section5Rule4.BuildingRule.SurfaceRule, self).__init__(
+                    rmrs_used=UserBaselineProposedVals(False, True, False),
+                    required_fields={},
+                )
+
+            def get_calc_vals(self, context, data=None):
+                surface = context.baseline
+                surface_conditioning_category_dict = data[
+                    "surface_conditioning_category_dict"
+                ]
+                surface_category = surface_conditioning_category_dict[surface["id"]]
+                surface_type = get_opaque_surface_type(surface)
+                return {
+                    "surface_category": surface_category,
+                    "surface_type": surface_type,
+                }
+
+            def applicability_check(self, context, calc_vals, data):
+                surface_category = calc_vals["surface_category"]
+                return surface_category != "UNREGULATED"
```

Comment:
Could you do the following at the beginning and then replace  "UNREGULATED" with ```SCC.UNREGULATED```?
```python
from rct229.rulesets.ashrae9012019.ruleset_functions.get_surface_conditioning_category_dict import (
    SurfaceConditioningCategory as SCC,
)